### DISCUSSION TITLE
How to properly update firebase packages?

### DISCUSSION THREAD

**Discussion by slavko-lukic** [invertase/react-native-firebase#7450] [Q&A] [answered]
I have already installed `messaging` and `app` packages version `18.5.0` from before.
Now I need to install `crashlytics` which is `18.6.1` currently.
Install fails because it's not matching with `app` package.
When I update `messaging` and `app` packages to 18.6.1 than `pod install` command fails, also suggested `pod install --repo-update` doesn't help.

I solved this problem by using `crashlytics@18.5.0` for now, but what is the proper way to update them all to new versions? I am using `npm`.

**Comment by mikehardy** [comment]
Our versioning philosophy: https://invertase.io/blog/react-native-firebase-versioning  (titled "One True Version" which gives you the general idea...)

You mention:

> When I update messaging and app packages to 18.6.1 than pod install command fails

...but you do not provide any error message to accompany the assertion - impossible to say what you need to do then, to use current versions (which are of course always recommended...)

**Reply by slavko-lukic** [reply]
@mikehardy My bad, I'll provide you more details.

1. I updated `package.json` versions of my both `app` and `messaging` packages from `18.5.0` to `18.6.1`
2. I ran `npm install`
3. Than I ran `pod install`

`pod install` gave following error: 
```console
[!] CocoaPods could not find compatible versions for pod "Firebase/CoreOnly":
  In snapshot (Podfile.lock):
    Firebase/CoreOnly (= 10.15.0)

  In Podfile:
    RNFBApp (from `../node_modules/@react-native-firebase/app`) was resolved to 18.6.1, which depends on
      Firebase/CoreOnly (= 10.17.0)


You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * changed the constraints of dependency `Firebase/CoreOnly` inside your development pod `RNFBApp`.
   You should run `pod update Firebase/CoreOnly` to apply changes you've made.
```

After that I ran `pod install --repo-update` which failed with error:
```console
[!] CocoaPods could not find compatible versions for pod "Firebase/CoreOnly":
  In snapshot (Podfile.lock):
    Firebase/CoreOnly (= 10.15.0)

  In Podfile:
    RNFBApp (from `../node_modules/@react-native-firebase/app`) was resolved to 18.6.1, which depends on
      Firebase/CoreOnly (= 10.17.0)


You have either:
 * changed the constraints of dependency `Firebase/CoreOnly` inside your development pod `RNFBApp`.
   You should run `pod update Firebase/CoreOnly` to apply changes you've made.
```

After that I ran `pod update Firebase/CoreOnly` which failed with error:
```console
[!] CocoaPods could not find compatible versions for pod "FirebaseCoreExtension":
  In snapshot (Podfile.lock):
    FirebaseCoreExtension (= 10.15.0)

  In Podfile:
    RNFBMessaging (from `../node_modules/@react-native-firebase/messaging`) was resolved to 18.6.1, which depends on
      FirebaseCoreExtension (= 10.17.0)


You have either:
 * changed the constraints of dependency `FirebaseCoreExtension` inside your development pod `RNFBMessaging`.
   You should run `pod update FirebaseCoreExtension` to apply changes you've made.
```

`pod update FirebaseCoreExtension` failed with error:
```console
[!] CocoaPods could not find compatible versions for pod "Firebase/Messaging":
  In snapshot (Podfile.lock):
    Firebase/Messaging (= 10.15.0)

  In Podfile:
    RNFBMessaging (from `../node_modules/@react-native-firebase/messaging`) was resolved to 18.6.1, which depends on
      Firebase/Messaging (= 10.17.0)


You have either:
 * changed the constraints of dependency `Firebase/Messaging` inside your development pod `RNFBMessaging`.
   You should run `pod update Firebase/Messaging` to apply changes you've made.
```

`pod update Firebase/Messaging` failed with error: 
```console
[!] CocoaPods could not find compatible versions for pod "FirebaseCoreExtension":
  In snapshot (Podfile.lock):
    FirebaseCoreExtension (= 10.15.0)

  In Podfile:
    RNFBMessaging (from `../node_modules/@react-native-firebase/messaging`) was resolved to 18.6.1, which depends on
      FirebaseCoreExtension (= 10.17.0)


You have either:
 * changed the constraints of dependency `FirebaseCoreExtension` inside your development pod `RNFBMessaging`.
   You should run `pod update FirebaseCoreExtension` to apply changes you've made.
```

`pod update FirebaseCoreExtension` failed with same error as above and it keeps looping from here.

**Reply by mikehardy** [reply, marked as answer]
So, delete Podfile.lock and rerun pod install ? It is supposed to be that easy. pod install is just telling you that your lockfile as it is, is incompatible with what's in your Podfile. Clear the lock and re-try.

**Reply by slavko-lukic** [reply]
Thanks. Wasn't sure if it was safe to delete podfile.lock.

**Reply by mikehardy** [reply]
It is usually safe - and if committed to your source control (it should be! I recommend it at least) then you always have a rollback
